src/log/log.c

```diff
@@ -976,7 +976,14 @@ __log_newfile(WT_SESSION_IMPL *session, bool conn_open, bool *created)
 	 * If we need to create the log file, do so now.
 	 */
 	if (create_log) {
-		log->prep_missed++;
+		/*
+		 * Do not increment missed prep logs counter when a hot backup
+		 * is in progress and pre-allocation is enabled since
+		 * we deliberately not using prep log files in that case
+		 * (see comments above).
+		 */
+		if ((conn->log_prealloc == 0) || !conn->hot_backup)
```

Comment:
Thank you @turchanov. I think the original fix is clearer (i.e. only check !conn->hot_backup in the conditional). Also, can you slightly modify the comment, as we generally use full, not abbreviated words and sentences in the comments. I suggest something like this:
"Only increment the missed pre-allocated file counter if a hot backup is not in progress. We are deliberately not using pre-allocated log files during backup (see comment above)."